app/controllers/api/users_controller.rb

```diff
@@ -0,0 +1,19 @@
+class API::UsersController < API::RestfulController
+  def index
+    instantiate_collection do |collection|
+      collection.where("users.name ilike :first OR
+                        users.name ilike :subsequent OR
+                        users.username ilike :first",
+                        first:       "#{params[:q]}%",
+                        subsequent:  "% #{params[:q]}%")
+    end
+    respond_with_collection serializer: Simple::UserSerializer
+  end
+
+  private
+  def accessible_records
+    resource_class.active.verified.distinct.
+    joins(:memberships).where('memberships.group_id': current_user.group_ids).
```

Comment:
This is duplicate logic of has_mentions#mentioned_users, which makes me think it wants to be a scope on User

```
# user.rb
scope :mentionable_by, (user) -> {
  active.verified.distinct.joins(:memberships).where("memberships.group_id": user.group_ids).where.not(id: user.id)
}
```

Reply:
nice